### ISSUE TITLE
Compatibility issue with relay-runtime 7.0.0?

### ISSUE THREAD

**Issue by nirvdrum** [relay-tools/react-relay-network-modern#81]
I'm trying to follow the example from the README, but TypeScript won't allow the example code to work with relay-runtime 7.0.0:

```
TypeScript error in /home/nirvdrum/dev/workspaces/my_project/src/components/Auth0RelayEnvironmentProvider.tsx(51,33):
Type 'RelayNetworkLayer' is not assignable to type 'Network'.
  Types of property 'execute' are incompatible.
    Type 'import("/home/nirvdrum/dev/workspaces/my_project/node_modules/react-relay-network-modern/node_modules/@types/relay-runtime/index").ExecuteFunction' is not assignable to type 'import("/home/nirvdrum/dev/workspaces/my_project/node_modules/@types/relay-runtime/lib/network/RelayNetworkTypes").ExecuteFunction'.
      Property 'concat' is missing in type 'import("/home/nirvdrum/dev/workspaces/my_project/node_modules/react-relay-network-modern/node_modules/@types/relay-runtime/index").Observable<import("/home/nirvdrum/dev/workspaces/my_project/node_modules/react-relay-network-modern/node_modules/@types/relay-runtime/index").GraphQLResponse>' but required in type 'import("/home/nirvdrum/dev/workspaces/my_project/node_modules/@types/relay-runtime/lib/network/RelayObservable").RelayObservable<import("/home/nirvdrum/dev/workspaces/my_project/node_modules/@types/relay-runtime/lib/network/RelayNetworkTypes").GraphQLResponse>'.  TS2322

    49 | 
    50 |   const store = new Store(new RecordSource())
  > 51 |   const env = new Environment({ network, store })
```

**Comment by ivosabev** [comment]
Any news on the 7 compatibility?

**Comment by nodkz** [comment]
Will be glad if community provide pull request.

Currently I have no bandwidth to do it myself.

**Comment by ivosabev** [comment]
I've just updated to 7.1 and all seems to be working fine.

**Comment by nirvdrum** [comment]
@ivosabev Did you add the package with `--ignore-optional`? If not, I don't see how they can be compatible. The latest release of this module (4.3.0) has [a hard-coded optional dependency](https://github.com/relay-tools/react-relay-network-modern/blob/v4.3.0/package.json#L35) on an old version of `@types/relay-runtime`.

I don't know what the right fix is here, otherwise I'd open a PR. I don't understand why there's an optional dependency at all. But as long as there is one, it's forever going to be in conflict with someone's `@types/relay-runtime` dependency.

**Comment by nirvdrum** [comment]
I've opened up a PR to remove the optional dependency. Does anyone in this thread require its presence? I can't see how removing it would break anyone's projects, but it'd be very helpful to know if it would.

**Comment by nirvdrum** [comment]
Since #85 was merged and v4.4.0 was released with that fix, this issue can be closed.
